File: TP_lab6/TP_lab6_3_Pogoda/TP_lab5_3_Pogoda/ViewController.m
//
//  ViewController.m
//  TP_lab5_3_Pogoda
//
//  Created by Admin on 14.04.17.
//  Copyright © 2017 Daniil Boyko. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@property (weak, nonatomic) IBOutlet UILabel *indicator;
@property (weak, nonatomic) IBOutlet UITextField *fildcity;
@property (weak, nonatomic) IBOutlet UISegmentedControl *inputtype;
@property (nonatomic, retain) NSMutableData *textData;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)refresh:(id)sender {
    NSURL *url;
    if ([[self inputtype] selectedSegmentIndex] == 0){
        NSString *city = [[self fildcity] text];
        url = [[NSURL alloc] initWithString: [NSString stringWithFormat:@"http://api.openweathermap.org/data/2.5/weather?q=%@&appid=2a3316ef209c91ac5c10a77e7717ac7b",city]];
    }
    else{
        NSArray *cordinats = [[[self fildcity] text] componentsSeparatedByString:@" "];
        url = [[NSURL alloc] initWithString: [NSString stringWithFormat:@"http://api.openweathermap.org/data/2.5/weather?lat=%@&lon=%@&appid=2a3316ef209c91ac5c10a77e7717ac7b",cordinats[0],cordinats[1]]];
    }
    //http://www.imaladec.com/story/downloading_data
    // создаем запрос //2 аргумент - это метод кэширования
    NSURLRequest *theRequest=[NSURLRequest requestWithURL:url
                                              cachePolicy:NSURLRequestUseProtocolCachePolicy
                                          timeoutInterval:60.0];
    
    NSURLConnection *theConnection = [NSURLConnection connectionWithRequest:theRequest delegate:self];
    if (theConnection) {
        self.textData = [NSMutableData data];
    } else {
        NSLog(@"Connection failed");
    }
    

}

- (void)connection:(NSURLConnection *)connection didReceiveData:(NSData *)data {
    [self.textData appendData:data];
}

- (void)connectionDidFinishLoading:(NSURLConnection *)connection {
    NSDictionary *forcasting = [NSJSONSerialization JSONObjectWithData: self.textData
                                                               options:NSJSONReadingMutableContainers error:nil];
    NSDictionary *maidictionary = [forcasting valueForKey:@"main"];
    NSNumber *temperature = [maidictionary valueForKey:@"temp"];
    NSNumber *result = [NSNumber numberWithDouble:[temperature doubleValue] - 273.15];
    
    if ([result doubleValue] <= 0)
        [[self indicator] setTextColor:[UIColor blueColor]];
    if ([result doubleValue] >= 0 && [result doubleValue] < 10 )
        [[self indicator] setTextColor:[UIColor yellowColor]];
    if ([result doubleValue] >= 10 && [result doubleValue] < 20 )
        [[self indicator] setTextColor:[UIColor orangeColor]];
    if ([result doubleValue] >=20 && [result doubleValue] )
        [[self indicator] setTextColor:[UIColor redColor]];
    
    
    if ([result doubleValue] > 0)
        [[self indicator] setText: [NSString stringWithFormat:@"+%2.1f °C",[result doubleValue]]];
    else
        [[self indicator] setText: [NSString stringWithFormat:@"%2.1f °C",[result doubleValue]]];
    
}

- (void)connection:(NSURLConnection *)connection didFailWithError:(NSError *)error {
    NSLog(@"%@", error);
    [[UIApplication sharedApplication] setNetworkActivityIndicatorVisible:NO];
}

@end

File: TP_lab8/TP_lab8_3/TP_lab8_3/ViewController.swift
//
//  ViewController.swift
//  TP_lab8_3
//
//  Created by Admin on 27.05.17.
//  Copyright © 2017 Daniil Boyko. All rights reserved.
//

import UIKit
import CoreData

class ViewController: UIViewController, UITableViewDataSource {

    
    var students: [NSManagedObject] = []
    @IBOutlet weak var studentNameTextField: UITextField!
    @IBOutlet weak var tableView: UITableView!
    @IBOutlet weak var sddStudentButton: UIButton!
    
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


    func numberOfSectionsInTableView(tableView: UITableView) -> Int {
        return 1
    }
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection: Int) -> Int {
        return students.count
    }
    
    func tableView(_ cellForRowAttableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "studentCell")! as UITableViewCell
        let student = students[indexPath.row]
        cell.textLabel?.text = student.value(forKey: "name") as? String //Заполняем текст ячейки таблицы значением ключа "name"
        return cell
    }
    
    func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCellEditingStyle, forRowAt indexPath: IndexPath) {
        if editingStyle == UITableViewCellEditingStyle.delete
        {
            let appDelegate = UIApplication.shared.delegate as! AppDelegate //Создаем ссылку на класс AppDelegate из файла AppDelegate.swift
            let managedObjectContext = appDelegate.managedObjectContext //Создаем ссылку на метод managedObjectContext из класса AppDelegate в файле AppDelegate.swift
            managedObjectContext!.delete(students[indexPath.row] as NSManagedObject) //Выбираем метод удаления объекта из модели данных students
            /*do
             {
             try managedObjectContext.save() //Пробуем сохранить изменения в базе данных
             students.removeAtIndex(indexPath.row) //Удаляем объект из модели students
             tableView.deleteRowsAtIndexPaths([indexPath], withRowAnimation: .Left) //Удаляем строку из таблицы
             }
             catch let error as NSError
             {
             print("Data removing error: \(error)") //В случае возникновения ошибки, выводим ее в консоль
             }*/
            var error: NSError? = nil
            do{
            managedObjectContext?.save()
            }
            catch(NSError, err){}
            if (error == nil)
            {
                students.remove(at: indexPath.row) //Удаляем объект из модели students
                tableView.deleteRows(at: [indexPath], with: .left) //Удаляем строку из таблицы
            }
            else
            {
                print("Data removing error: \(String(describing: error))") //В случае возникновения ошибки, выводим ее в консоль
            }
        }
    }
    
    @IBAction func addStudentBut(_ sender: Any) {
    }

}

